src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/GeneralizedCoastlineCheck.java

```diff
@@ -0,0 +1,111 @@
+package org.openstreetmap.atlas.checks.validation.linear.lines;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.Segment;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.LineItem;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.tags.NaturalTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Flags generalized coastlines-- that is, OSM ways with the tag natural=coastline where x% or more
+ * node pairs are y or more meters apart. X and y can be specified in a {@link Configuration}.
+ * Coastlines can be represented in Atlas by {@link LineItem}s and may be members of
+ * {@link Relation}s.
+ *
+ * @author seancoulter
+ */
+public class GeneralizedCoastlineCheck extends BaseCheck<Long>
+{
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            "This coastline is generalized, as {0}% of node pairs are {1} or more meters apart. To fix, increase the number of nodes along this coastline.");
+    private static final long MINIMUM_DISTANCE_BETWEEN_NODES = 100;
+    private static final double MINIMUM_NODE_PAIR_THRESHOLD_PERCENTAGE = 30.0;
+
+    private static final double PERCENTAGE_ADJUST = 100.0;
+    private final double percentageThreshold;
+    private final long minimumDistanceBetweenNodes;
+
+    private static final Logger logger = LoggerFactory.getLogger(GeneralizedCoastlineCheck.class);
+
+    public GeneralizedCoastlineCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.percentageThreshold = this.configurationValue(configuration, "node.minimum.threshold",
+                MINIMUM_NODE_PAIR_THRESHOLD_PERCENTAGE);
+        this.minimumDistanceBetweenNodes = this.configurationValue(configuration,
+                "node.minimum.distance", MINIMUM_DISTANCE_BETWEEN_NODES);
+    }
+
+    /**
+     * This method validates or invalidates the supplied atlas object for the check
+     *
+     * @param object
+     *            The {@link AtlasObject} being checked
+     * @return True if the {@link AtlasObject} is a {@link LineItem} with the tag natural=coastline,
+     *         or has a parent relation with that tag; false otherwise
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof LineItem
```

Comment:
Have you come across any examples of coastline relations with edges? Is that the reason why we are going for LineItem instead of Line?

Reply:
I haven't yet, but I wanted to keep the filter general in case we do run into that. I've looked at Iceland and some of Argentina so far, but my thinking is that maybe other countries may record Edges/Lines differently.